test/tracing/ioredis/app.js

```diff
@@ -65,6 +68,51 @@ app.get('/values', function(req, res) {
 });
 
 
+app.get('/keepTracing', function(req, res) {
+  var key = req.query.key;
+  var redisResponse = null;
+  client.get(key).then(function(redisRes) {
+    redisResponse = redisRes;
+    // Execute another traced call to verify that we keep the tracing context.
+    return request('http://127.0.0.1:' + agentPort);
+  }).then(function(httpRes) {
+    res.send(httpRes + ';' + redisResponse);
+  }).catch(function(err) {
+    log('Unexpected error for key %s', key, err);
+    res.sendStatus(500);
+  });
+});
+
+
+app.get('/keepTracingCallback', function(req, res) {
+  // this uses a self created promise and a mix promise and callback styles,
+  // in particular, it uses the ioredis optional callback argument.
+  var key = req.query.key;
+  return new Promise(function(resolve, reject) {
+    // using ioredis client with callback instead of a promise here
+    client.get(key, function(err, redisRes) {
+      if (err) {
+        log('Get with key %s failed', key, err);
+        return reject(err);
+      }
+      // Execute another traced call to verify that we keep the tracing context.
+      request('http://127.0.0.1:' + agentPort, function(httpErr, httpRes) {
+        if (httpErr) {
+          log('HTTP call failed', httpErr);
+          return reject(httpErr);
+        }
+        return resolve(httpRes.body + ';' + redisRes);
+      });
+    });
+  }).then(function(result) {
+    res.send(result);
```

Comment:
How about we send information about the current tracing context as part of the HTTP response to the tests? Then we could verify that we have the correct span context in this location (which should be HTTP entry).

Could work similar to this:

https://github.com/instana/nodejs-sensor/blob/master/test/tracing/bluebird/app.js#L101
https://github.com/instana/nodejs-sensor/blob/master/test/tracing/bluebird/test.js#L70-L74